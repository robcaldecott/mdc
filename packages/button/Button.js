import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const Button = ({ className, raised, compact, dense, children, ...other }) => (
  <button
    className={classnames(
      "mdc-button",
      {
        "mdc-button--raised": raised,
        "mdc-button--compact": compact,
        "mdc-button--dense": dense
      },
      className
    )}
    {...other}
  >
    {children}
  </button>
);

Button.propTypes = {
  raised: PropTypes.bool,
  compact: PropTypes.bool,
  dense: PropTypes.bool
};

export default Button;
