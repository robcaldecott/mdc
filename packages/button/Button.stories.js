import React from "react";
import { Button } from ".";
import { storiesOf } from "@storybook/react";
import { withKnobs, text, boolean } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { withInfo } from "@storybook/addon-info";

storiesOf("button", module)
  .addDecorator(withKnobs)
  .add(
    "props",
    withInfo({ header: false, inline: true })(() => (
      <Button
        raised={boolean("raised", false)}
        dense={boolean("dense", false)}
        unelevated={boolean("unelevated", false)}
        outlined={boolean("outlined", false)}
        disabled={boolean("disabled", false)}
        icon={boolean("icon", false) ? "star" : ""}
        {...(boolean("Custom")
          ? { style: { "--mdc-theme-primary": "red" } }
          : {})}
        onClick={action("onClick")}
      >
        {text("Label", "Button")}
      </Button>
    ))
  );
