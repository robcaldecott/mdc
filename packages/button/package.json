{
  "name": "@mdc/button",
  "version": "0.29.0",
  "description": "React MDC Button",
  "main": "index.js",
  "author": "Rob Caldecott <robert.caldecott@gmail.com>",
  "license": "MIT",
  "dependencies": {
    "classnames": "^2.2.5",
    "prop-types": "^15.6.0"
  }
}
