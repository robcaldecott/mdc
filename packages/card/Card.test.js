import React from "react";
import {
  Card,
  CardPrimary,
  CardTitle,
  CardSubtitle,
  CardSupportingText,
  CardActions,
  CardAction
} from ".";
import { Button } from "@mdc/button";
import renderer from "react-test-renderer";

it("renders", () => {
  expect(
    renderer.create(
      <Card>
        <CardPrimary>
          <CardTitle large>Title</CardTitle>
          <CardSubtitle>Subtitle</CardSubtitle>
        </CardPrimary>
        <CardSupportingText>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do
          eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
          minim veniam, quis nostrud exercitation ullamco laboris nisi ut
          aliquip ex ea commodo consequat.
        </CardSupportingText>
        <CardActions>
          <CardAction>
            <Button compact>Action 1</Button>
          </CardAction>
          <CardAction>
            <Button compact>Action 2</Button>
          </CardAction>
        </CardActions>
      </Card>
    )
  ).toMatchSnapshot();
});
