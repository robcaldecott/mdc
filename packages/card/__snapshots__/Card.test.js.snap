// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="mdc-card"
>
  <section
    className="mdc-card__primary"
  >
    <h1
      className="mdc-card__title mdc-card__title--large"
    >
      Title
    </h1>
    <h2
      className="mdc-card__subtitle"
    >
      Subtitle
    </h2>
  </section>
  <section
    className="mdc-card__supporting-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
  </section>
  <section
    className="mdc-card__actions"
  >
    <button
      className="mdc-button mdc-button--compact mdc-card__action"
    >
      Action 1
    </button>
    <button
      className="mdc-button mdc-button--compact mdc-card__action"
    >
      Action 2
    </button>
  </section>
</div>
`;
